src/backend/distributed/utils/node_metadata.c

```diff
@@ -478,10 +529,17 @@ RemoveNodeFromCluster(char *nodeName, int32 nodePort)
 		deletedNodeId = workerNode->nodeId;
 	}
 
-	DeleteNodeRow(nodeName, nodePort);
-
 	DeleteAllReferenceTablePlacementsFromNode(nodeName, nodePort);
 
+	hasAnyShardPlacements = NodeHasShardPlacements(nodeName, nodePort,
+												   onlyConsiderActivePlacements);
+	if (hasAnyShardPlacements)
+	{
+		ereport(ERROR, (errmsg("you cannot remove a node which has shard placements")));
```

Comment:
Why consider inactive placements? Also, Were we planning to delete inactive shard placements?